web-ui/src/app/components/tree-visualizer/tree-visualizer.js

```diff
@@ -446,41 +447,60 @@ var jQuery = require('jquery');
                 var targetLink = tree.find("a.hovered");
                 if (targetLink.length) {
                     var tooltip = tree.next(".tv-tooltip"),
+                        arrow = tooltip.siblings(".arrow"),
                         targetRect = targetLink[0].getBoundingClientRect(),
                         treeRect = tree[0].getBoundingClientRect(),
+                        arrowRect = arrow[0].getBoundingClientRect(),
                         targetV = {
-                            right: $window.outerWidth() - targetRect.right,
-                            bottom: $window.outerHeight() - targetRect.bottom,
+                            right: $window.width() - targetRect.right,
+                            bottom: $window.innerHeight() - targetRect.bottom,
                         },
                         treeV = {
-                            right: $window.outerWidth() - treeRect.right,
-                            bottom: $window.outerHeight() - treeRect.bottom,
+                            right: $window.width() - treeRect.right,
+                            bottom: $window.innerHeight() - treeRect.bottom,
                         };
-
+                    // The maximum to the right the tooltip can be positioned without overlapping the viewport
+                    var rightestLeft = parseInt($window.width() - tooltip.outerWidth() - 20, 10)
```

Comment:
Building on my comment below, instead of `$window.width()` perhaps something like `Math.max($window.width(), /* right most point of the node */)`